fuel-core/src/database/utils.rs

```diff
@@ -0,0 +1,218 @@
+use crate::{
```

Comment:
I'd prefer to avoid a generic `utils` module, as it makes organization less clear and will just keep growing with potentially unrelated things. 

Since this is mostly about coins and messages, maybe rename this utils module into `resources` within the database instead?